src/main/java/com/minecolonies/coremod/quests/QuestManager.java

```diff
@@ -269,4 +278,32 @@ public void deserializeNBT(final CompoundTag nbt)
         }
         this.questReputation = nbt.getDouble(TAG_REPUTATION);
     }
+
+    @Override
+    public List<IQuestInstance> getAvailableQuests()
+    {
+        return new ArrayList<>(availableQuests.values());
+    }
+
+    @Override
+    public List<IQuestInstance> getInProgressQuests()
+    {
+        return new ArrayList<>(inProgressQuests.values());
+    }
+
+    @Override
+    public List<FinishedQuest> getFinishedQuests()
+    {
+        if (finishedQuestsCache == null)
+        {
+            List<FinishedQuest> data = new ArrayList<>();
+            for (Map.Entry<ResourceLocation, Integer> entry : finishedQuests.entrySet())
```

Comment:
Im still unsure about this. As we only query this on the client side, I think we can build this in the window on demand just as well